runelite-client/src/main/java/net/runelite/client/plugins/npchighlight/NpcIndicatorsPlugin.java

```diff
@@ -270,16 +267,47 @@ public void onMenuEntryAdded(MenuEntryAdded event)
 		}
 		else if (menuAction == MenuAction.EXAMINE_NPC && client.isKeyPressed(KeyCode.KC_SHIFT))
 		{
-			// Add tag option
+			// Add tag and tag-all options
+			final int id = event.getIdentifier();
+			final NPC[] cachedNPCs = client.getCachedNPCs();
+			final NPC npcCached = cachedNPCs[id];
```

Comment:
```suggestion
			final NPC npc = cachedNPCs[id];
```

The "cached" name the API uses is basically mis-named currently.